spec/Classes.tex

```diff
@@ -784,12 +787,54 @@ \subsection{The Method Receiver and the {\em this} Argument}
 is \chpl{c1}.
 \end{chapelexample}
 
-The optional \sntx{this-intent} declares the intent by which the receiver
-argument will be passed.  When the value of \sntx{this-intent} is \chpl{param},
-it implies that the function can only be applied to param objects of the given
-type binding.  If the value is \chpl{ref}, the argument will be passed by
-reference---allowing modification of \chpl{this}.  Otherwise, the argument will
-be passed with the blank intent as specified in \rsec{The_Blank_Intent}.
+The optional \sntx{this-intent} declares how the receiver argument is
+constrained or will be passed for a given method call.
+
```

Comment:
I may be misinterpreting the paragraph starting on 793 but I think it mis-states the notion of static
member functions in C++ and so then I am uncertain what is intended for Chapel.  In C++ it is, as
far as I know, valid to invoke a static method on an instance of a class.

Foo item(1, 2, 4);

item.staticMethod(10);

is the same as Foo::staticMethod(10);


I wonder if it is a pro or a con to mention C++ "static methods" esp. if the behavior is intended
to be a little different.  "static" is a bit funny in C/C++ since it means several different things.
Surely in C++ one can even define a "static local variable" inside a "static method" and have that
static method call a file-static function. (Haven't compiled to verify).